spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -555,10 +560,13 @@ public void testDeleteWithScalarSubquery() throws NoSuchTableException {
 
     createOrReplaceView("deleted_id", Arrays.asList(1, 100, null), Encoders.INT());
 
-    sql("DELETE FROM %s t WHERE id <= (SELECT min(value) FROM deleted_id)", tableName);
-    assertEquals("Should have expected rows",
-        ImmutableList.of(row(2, "hardware"), row(null, "hr")),
-        sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", tableName));
+    // TODO: Spark does not support AQE and DPP with aggregates at the moment
```

Comment:
That's an issue I don't have a solution for. It will be pretty tough if even possible to overcome.

Reply:
What happens when adaptive execution is not disabled?

Reply:
It leads to a runtime exception.

```
class org.apache.spark.sql.catalyst.plans.logical.Aggregate cannot be cast to class org.apache.spark.sql.execution.SparkPlan
```

I did debug it at some point. If I remember correctly, it was related to the fact that dynamic subqueries in Spark cannot have other subqueries or aggregates.